Kernel/Net/NetworkTask.cpp

```diff
@@ -276,7 +280,25 @@ void handle_udp(const IPv4Packet& ipv4_packet, const Time& packet_timestamp)
     auto& destination = ipv4_packet.destination();
 
     if (destination == IPv4Address(255, 255, 255, 255) || NetworkAdapter::from_ipv4_address(destination) || socket->multicast_memberships().contains_slow(destination))
-        socket->did_receive(ipv4_packet.source(), udp_packet.source_port(), KBuffer::copy(&ipv4_packet, sizeof(IPv4Packet) + ipv4_packet.payload_size()), packet_timestamp);
+        socket->did_receive(ipv4_packet.source(), udp_packet.source_port(), { &ipv4_packet, sizeof(IPv4Packet) + ipv4_packet.payload_size() }, packet_timestamp);
+}
+
+void queue_tcp_ack(RefPtr<TCPSocket> socket)
+{
+    if (!delayed_ack_sockets)
+        delayed_ack_sockets = new HashTable<RefPtr<TCPSocket>>;
+    delayed_ack_sockets->set(move(socket));
+}
+
+void flush_tcp_acks()
+{
+    if (!delayed_ack_sockets)
+        return;
+    for (auto& socket : *delayed_ack_sockets) {
+        Locker locker(socket->lock());
+        [[maybe_unused]] auto result = socket->send_tcp_packet(TCPFlags::ACK);
```

Comment:
What if send_tcp_packet fails, do we not care? What should we do if this background thread gets `EHOSTUNREACH` for example...?

Reply:
Well, there really isn't much we can do. If the adapter/route disappears the TCP connection's lifecycle is pretty much over anyway.